File: typescipt/existing/src/language/expressions/functions/yearFunction.ts



namespace Lexy.Compiler.Language.Expressions.Functions;

public class YearFunction : SingleArgumentFunction
{
   public const string Name = "YEAR";

   protected override string FunctionHelp => $"'{Name} expects 1 argument (Date)";

   protected override VariableType ArgumentType => PrimitiveType.Date;
   protected override VariableType ResultType => PrimitiveType.Number;

   private YearFunction(Expression valueExpression, SourceReference reference)
     : base(valueExpression, reference)
   {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression expression)
   {
     return new YearFunction(expression, reference);
   }
}

File: src/Lexy.Compiler/Language/Expressions/Functions/YearFunction.cs
using Lexy.Compiler.Language.VariableTypes;

namespace Lexy.Compiler.Language.Expressions.Functions;

public class YearFunction : SingleArgumentFunction
{
    public const string Name = "YEAR";

    protected override string FunctionHelp => $"'{Name} expects 1 argument (Date)";

    private YearFunction(Expression valueExpression, ExpressionSource source)
        : base(Name, valueExpression, source, PrimitiveType.Date, PrimitiveType.Number)
    {
    }

    public static FunctionCallExpression Create(ExpressionSource source, Expression expression)
    {
        return new YearFunction(expression, source);
    }
}